### PR TITLE
Add minimal metrics process template for Windows

### PR DESCRIPTION
**Pull request by andrwils**

### CODE REVIEW

File: config/templates/metric_process_windows/filter/filters.conf.njk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Add service.environment {{ environment }}{% endif %}
+    Add process.name ${AGENT_NAME}
+    Add @metadata.keyAsPath true
+
+[FILTER]
+    Name     throttle
+    Match    {{typeTag}}
+    Rate     1
+    Window   1
+    Interval 1m
```
Comment (mbystedt) [inline config/templates/metric_process_windows/filter/filters.conf.njk:22]: {{inputIntervalSec}}s

File: config/templates/metric_process_windows/filter/filters.conf.njk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Add service.name {{ component | lower }}{% if environment %}
+    Add service.environment {{ environment }}{% endif %}
+    Add process.name ${AGENT_NAME}
+    Add @metadata.keyAsPath true
+
+[FILTER]
+    Name     throttle
+    Match    {{typeTag}}
+    Rate     1
+    Window   1
```
Comment (mbystedt) [inline config/templates/metric_process_windows/filter/filters.conf.njk:21]: Docs seem to indicate a larger window may smooth things.

File: config/server/stress.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
         "component": "as-as-api",
         "tmplTomcatTestApplicationIndex": "nrm-test-aw-<%=YYYY.MM=%>"
       }
-    }
+    },
+    {"id": "metric_process_windows", "type": "metric_process_windows", "context": {}}
```
Comment (mbystedt) [inline config/server/stress.json:19]: Move to gen.ts#64

**Review by mbystedt — CHANGES_REQUESTED** [review]

File: config/templates/metric_process_windows/filter/filters.conf.njk (hunk 1 above)
Comment (andrwils) [reply config/templates/metric_process_windows/filter/filters.conf.njk]: Done

File: config/templates/metric_process_windows/filter/filters.conf.njk (hunk 2 above)
Comment (andrwils) [reply config/templates/metric_process_windows/filter/filters.conf.njk]: Set to 10

File: config/server/stress.json (hunk 3 above)
Comment (andrwils) [reply config/server/stress.json]: Done

File: src/commands/gen.ts [hunk 4]
```diff
@@ -53,15 +53,24 @@ export default class Gen extends Command {
...
-              'childProcess': true,
-            },
-          }, serverConfig, flags.context);
+          if (serverConfig.os === 'linux') {
+            serviceArr[agentCount].writeApp({
+              id: `metrics_fluentbit_process`,
+              type: 'metric_s6_process',
+              context: {
+                'app': 'fluentbit',
+                'component': `fluent-bit.${agentCount}`,
```
Comment (andrwils) [inline src/commands/gen.ts:62]: Should metric_process_windows also have these context settings?

Comment (mbystedt) [reply src/commands/gen.ts:62]: Yes

File: src/commands/gen.ts [hunk 5]
```diff
@@ -53,15 +53,24 @@ export default class Gen extends Command {
...
-          serviceArr[agentCount].writeApp({
-            id: `metrics_fluentbit_process`,
-            type: 'metric_s6_process',
-            context: {
-              'app': 'fluentbit',
-              'component': `fluent-bit.${agentCount}`,
-              'childProcess': true,
-            },
-          }, serverConfig, flags.context);
+          if (serverConfig.os === 'linux') {
```
Comment (mbystedt) [inline src/commands/gen.ts:56]: This can be removed. writeApp will not output apps for the wrong OS.

**Review by mbystedt — CHANGES_REQUESTED** [review]

**Review by mbystedt — APPROVED** [review]

**Pull request merged**